### PR TITLE
{ARO} Fix integration tests for aro subcommand

### PR DESCRIPTION
**Pull request by petrkotas**
**Description**  
The ARO integration testing was failing due to unstable UUID  for [role assingment](https://github.com/Azure/azure-cli/blob/b38e150c5d41f80859b329612f8cd70c9d1efc2a/src/azure-cli/azure/cli/command_modules/aro/_rbac.py#L39).

The issue is solved using mock GUID generation.

**Testing Guide**  

**History Notes**  

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S169

**Comment by MyronFanQiu** [comment]
@petrkotas To overcome the uuid issue, you could take a look at this example.
https://github.com/Azure/azure-cli/blob/b38e150c5d41f80859b329612f8cd70c9d1efc2a/src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_general_operations.py#L43

**Comment by petrkotas** [comment]
@MyronFanQiu thank you for the pointer. I have fixed it. It seems to be working now.

File: src/azure-cli/azure/cli/command_modules/aro/tests/latest/test_aro_commands.py [hunk 1]
```diff
@@ -95,130 +29,311 @@ def test_aro_show(self, resource_group):
...
+
+        self.kwargs.update(
+            {
+                'master_subnet_resource': master_subnet_id,
+                'worker_subnet_resource': worker_subnet_id,
+            }
+        )
+
+        # test aro create
+        with mock.patch('azure.cli.command_modules.aro._rbac._gen_uuid', side_effect=self.create_guid):
```
Comment (m1kola) [inline src/azure-cli/azure/cli/command_modules/aro/tests/latest/test_aro_commands.py:58]: Why do we have so many changes? I was expecting only `aro create` be wrapped with a mock. Could you please explain rest of the changes?

File: src/azure-cli/azure/cli/command_modules/aro/_rbac.py [hunk 2]
```diff
@@ -14,8 +14,10 @@
 
 CONTRIBUTOR = 'b24988ac-6180-42a0-ab88-20f7382dd24c'
 
+def _gen_uuid():
+    return uuid.uuid4()
```
[the remark this answers is not in the document]
Comment (petrkotas) [reply src/azure-cli/azure/cli/command_modules/aro/_rbac.py:18]: I have introduced the salt originally to solve the stability issue to allow test recordings. However, this has proven to be not helping. 

To solve the issue properly, I was given the advice to use mock UID generation, as is used in the rest of the `az` tool.

To my best knowledge, the random UID, which is present when not in recording does not pose any issue for the production use.

Am I right? @mjudeikis

File: src/azure-cli/azure/cli/command_modules/aro/tests/latest/test_aro_commands.py (hunk 1 above)
Comment (petrkotas) [reply src/azure-cli/azure/cli/command_modules/aro/tests/latest/test_aro_commands.py:58]: The big amount of changes goes on my head. I decided to group changes with the second issue the ARO test has. 

Having the `aro create` in every test is tedious and requires quite a lot of computing power and money.

Therefore I have collected all tests in the single test function and sorted them in order so they can be safely tested.

Thus only one `aro create` is required.

Comment (petrkotas) [reply src/azure-cli/azure/cli/command_modules/aro/tests/latest/test_aro_commands.py:58]: The comment leftovers are going to be removed as soon as I see the CI results.

**Comment by mjudeikis** [comment]
@m1kola can you please check and lgtm or not? I'm not feeling comfortable with python that much?

File: src/azure-cli/azure/cli/command_modules/aro/_rbac.py [hunk 3]
```diff
@@ -36,7 +40,7 @@ def assign_contributor_to_vnet(cli_ctx, vnet, object_id, label):
 
     # use object_id as a namespace and vnet and label as a salt for generating UUID to
     # make it stable across calls
```
Comment (m1kola) [inline src/azure-cli/azure/cli/command_modules/aro/_rbac.py:42]: The comment is no longer valid?

**Comment by petrkotas** [comment]
@m1kola thanks for the review! I have removed the comment, you were right. It was confusing leftover.

**Comment by ehashman** [comment]
@qwordy can we get this merged? Looks like ARO tests are still flaking on HEAD, did not realize this had been sitting :sweat_smile:

**Comment by qwordy** [comment]
> 
> 
> @qwordy can we get this merged? Looks like ARO tests are still flaking on HEAD, did not realize this had been sitting 😅

Sure

**Pull request merged**